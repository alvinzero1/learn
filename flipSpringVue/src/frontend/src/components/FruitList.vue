<!-- FruitList.vue -->
<script setup>
import { computed } from '@vue/reactivity';
import { ref } from 'vue'

const fruits = ref(["apple", "banana", "orange"])

const numFruits = computed(() => {
  return fruits.value.length
})

function addFruit(fruit) {
  fruits.value.push(fruit)
}

</script>

<template>
  <div>
    <ol>
      <li v-for="fruit in fruits" :key="fruit">
        {{ fruit }}
      </li>
    </ol>
    <p>
      Number of fruits: {{ numFruits }}
    </p>
  </div>
</template>