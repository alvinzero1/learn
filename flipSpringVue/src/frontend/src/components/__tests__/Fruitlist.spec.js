//  https://blog.canopas.com/vue-3-component-testing-with-jest-8b80a8a8946b
import { describe, test, expect } from 'vitest' // <<  define `test`
import { shallowMount } from '@vue/test-utils'
import FruitList from '@/components/fruitList.vue'

describe('FruitList component test', () => {



    test("tests data attributes", () => {
        const wrapper = shallowMount(FruitList)
        expect(wrapper.vm.fruits).toEqual(["apple", "banana", "orange"]);
    })

    test('return total number of fruits', () => {
        const wrapper = shallowMount(FruitList)
        expect(wrapper.vm.numFruits).toBe(3);
    })

    test('displays a list of fruits', () => {
        const wrapper = shallowMount(FruitList)
        const fruits = wrapper.findAll('li')

        expect(fruits.at(0).text()).toBe('apple')
        expect(fruits.at(1).text()).toBe('banana')
        expect(fruits.at(2).text()).toBe('orange')
    })

    test('displays the number of fruits', () => {
        const wrapper = shallowMount(FruitList)
        const numFruits = wrapper.find('p')
        expect(numFruits.text()).toBe('Number of fruits: 3')
    })

    test('tests addFruit method', () => {
        const wrapper = shallowMount(FruitList)

        const vm = wrapper.vm
        expect(vm.fruits.length).toBe(3)

        // add mango to the fruit list
        vm.addFruit('mango')
        console.log("> FruitList.fruits > " + vm.numFruits)

        expect(vm.fruits.length).toBe(4)
    })

    // Need tp quit and restart to update
    test('displays of fruits', () => {
        const wrapper = shallowMount(FruitList)
        console.log("> html > " + wrapper.html())
        expect(wrapper.html()).toMatchSnapshot() // <<<<< snapshot
    })

    test('my test all', () => {
        const wrapper = shallowMount(FruitList)
        const vm = wrapper.vm
        expect(vm.fruits).toEqual(["apple", "banana", "orange"])
        expect(vm.numFruits).toBe(3)
        vm.addFruit('mango')
        expect(vm.fruits.length).toBe(4)

        vm.fruits = (["aaa", "bbb"])
        console.log("> vm.fruits > " + vm.fruits)
        expect(vm.fruits).toEqual(["aaa", "bbb"])
        expect(vm.numFruits).toBe(2)
    })

})