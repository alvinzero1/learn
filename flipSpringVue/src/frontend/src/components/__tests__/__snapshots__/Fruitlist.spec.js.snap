// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FruitList component test > displays of fruits 1`] = `
"<div>
  <ol>
    <li>apple</li>
    <li>banana</li>
    <li>orange</li>
  </ol>
  <p> Number of fruits: 3</p>
</div>"
`;
